### PR TITLE
Cleanup + support functional components

### PR DESCRIPTION
**Pull request by skovhus**
My goals: 
- make functional components work (existing code can do most of it already)
- remove all the dead code (and there is quite a bit)
- better test coverage (+ fix smaller bugs)
- make README clearer on how to run this (currently there are multiple transformers, which can be confusing)
- clean up tests to use Jest snapshots
- utilize ReactUtils where possible

FYI @billyvg Are you ok with me doing a rather drastic cleanup/refactoring?

### CODE REVIEW

**Comment by billyvg** [comment]
@skovhus Yep, totally ok with it

**Comment by DanielMSchmidt** [comment]
@skovhus What is the state on this? Can you describe cases that break it so that I may resume the work on it?

**Comment by skovhus** [comment]
It's (almost) done. The handling is a lot better now... so it should be ready for merge now. @billyvg will you look into this? : )

/Kenneth

**Comment by skovhus** [comment]
@billyvg this should be ready for you to review and test now. 
- Functional component is now supported
- the code has been cleaned up a lot
- it now handles having multiple components in one file
- there is only one transformer (will not touch code without React)
- simpler test structure using Jest Snapshots
- coverage

**Comment by DanielMSchmidt** [comment]
Tested it, worked pretty good, just one annotation:

This is the current behavior
```diff
- function Foo({a,b}) {
+ function Foo(props: Props) {
+   const {a,b} = props;
    ...
}
```

I would love to see something like this instead:

```diff
- function Foo({a,b}) {
+ function Foo({a,b}: Props) {
    ...
}
```

File: src/transformers/functional.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  const param = params[0];
+
+  const newTypeAnnotation = j.typeAnnotation(
+    j.genericTypeAnnotation(j.identifier(typeIdentifier), null)
+  );
+
+  if (param.type === 'Identifier') {
+    param.typeAnnotation = newTypeAnnotation;
+  } else if (param.type === 'ObjectPattern') {
+    // NOTE: something is wrong with recast and objectPatterns...
```
Comment (skovhus) [inline src/transformers/functional.js]: @billyvg this is the reason for the spreads inside the body...

Comment (skovhus) [inline src/transformers/functional.js]: Didn't have type to investigate this more thoroughly, but seems to be a limitation in recast.

**Comment by skovhus** [comment]
@DanielMSchmidt thanks for testing it out... The problems is related to Recast, see my comments in the code and here https://github.com/benjamn/recast/issues/368

**Comment by billyvg** [comment]
Thanks! I will review this weekend

**Comment by skovhus** [comment]
Thanks. 😄🙌🏻

**Comment by billyvg** [comment]
No, thank you!

**Comment by skovhus** [comment]
No problem. Happily used the codemod several times over last last couple of months. : )

**Pull request merged**